### PR TITLE
VL-0 | added project skeleton

### PR DESCRIPTION
**Pull request by volod**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,104 @@
 *.class
 *.log
+
+# sbt specific
+dist/*
+target/
```
Comment (SuperIzya) [inline .gitignore:6]: `**/target` - for targets in 'project', 'project/project' etc.

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  .settings(volalandSettings: _*)
+  .settings(
+    addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.4"),
+    scalacOptions += "-Ypartial-unification"
+  )
+  .settings(libraryDependencies ++= Seq(
+    "org.typelevel"    %% "cats-core"      % catsCoreVersion,
+    "org.typelevel"    %% "cats-effect"    % catsEffectVersion,
+    "org.typelevel"    %% "cats-mtl-core"  % catsMtlVersion,
+    "org.typelevel"    %% "alleycats-core" % catsCoreVersion,
```
Comment (SuperIzya) [inline build.sbt:67]: What about cats.laws and cats.test in Test?

File: cats/src/scala/land/vola/cats/SparkTasks.scala [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package volaland
+package cats
+
+import _root_.cats.Id
+import _root_.cats.data.Kleisli
+import org.apache.spark.SparkContext
+
+object SparkTask {
+  def apply[A](f: SparkContext => A): SparkTask[A] =
+    Kleisli[Id, SparkContext, A](f)
```
Comment (SuperIzya) [inline cats/src/scala/land/vola/cats/SparkTasks.scala:10]: Why not use SparkTask[A] here?

File: cats/src/scala/land/vola/cats/SparkTasks.scala [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  def pure[A](a: => A): SparkTask[A] =
+    Kleisli[Id, SparkContext, A](_ => a)
```
Comment (SuperIzya) [inline cats/src/scala/land/vola/cats/SparkTasks.scala:13]: And here...

File: scalastyle-config.xml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+ <check level="warning" class="org.scalastyle.scalariform.SpacesBeforePlusChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.file.FileLineLengthChecker" enabled="true">
+  <parameters>
+   <parameter name="maxLineLength"><![CDATA[160]]></parameter>
+   <parameter name="tabSize"><![CDATA[4]]></parameter>
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.ClassNamesChecker" enabled="true">
+  <parameters>
+   <parameter name="regex"><![CDATA[[A-Z][A-Za-z]*]]></parameter>
```
Comment (SuperIzya) [inline scalastyle-config.xml:39]: No numbers in class names?

File: scalastyle-config.xml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.ObjectNamesChecker" enabled="true">
+  <parameters>
+   <parameter name="regex"><![CDATA[[A-Z][A-Za-z]*]]></parameter>
```
Comment (SuperIzya) [inline scalastyle-config.xml:44]: No numbers in object names either((

File: scalastyle-config.xml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.PackageObjectNamesChecker" enabled="true">
+  <parameters>
+   <parameter name="regex"><![CDATA[^[a-z][A-Za-z]*$]]></parameter>
```
Comment (SuperIzya) [inline scalastyle-config.xml:49]: For package objects i don't really see any use in numbers in names...

File: scalastyle-config.xml [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.MethodLengthChecker" enabled="true">
+  <parameters>
+   <parameter name="maxLength"><![CDATA[50]]></parameter>
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.MethodNamesChecker" enabled="true">
+  <parameters>
+   <parameter name="regex"><![CDATA[^[a-z][A-Za-z0-9]*$]]></parameter>
```
Comment (SuperIzya) [inline scalastyle-config.xml:106]: Thank you for allowing numbers in method names

File: scalastyle-config.xml [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.NumberOfMethodsInTypeChecker" enabled="true">
+  <parameters>
+   <parameter name="maxMethods"><![CDATA[30]]></parameter>
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.PublicMethodsHaveTypeChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.file.NewLineAtEofChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.file.NoNewLineAtEofChecker" enabled="false"></check>
+</scalastyle>
```
Comment (SuperIzya) [inline scalastyle-config.xml:117]: No new line!

File: build.sbt (hunk 2 above)
Comment (volod) [reply build.sbt:67]: now there are minimal dependencies that will be used in some way. 
let's add dependencies in moment when they really used.

File: .gitignore (hunk 1 above)
Comment (volod) [reply .gitignore:6]: means? this sbt config generate multi project target builds (core/targets etc), and it is logically exclude all the builds

File: cats/src/scala/land/vola/cats/SparkTasks.scala (hunk 3 above)
Comment (volod) [reply cats/src/scala/land/vola/cats/SparkTasks.scala:10]: I did't understand proposal, it's standard apply factory (typed, not companion object). Where exactly do you want type?

File: scalastyle-config.xml (hunk 5 above)
Comment (volod) [reply scalastyle-config.xml:39]: Yes, again it's pretty standard generated with scalastyle rules. Imo only alphabet class name make sence for readability. CamelCase.

File: scalastyle-config.xml (hunk 6 above)
Comment (volod) [reply scalastyle-config.xml:44]: It's logical, that companion objects have the same naming rules as classes.

File: .gitignore (hunk 1 above)
Comment (SuperIzya) [reply .gitignore:6]: `target/` includes all appearances of target folder?

File: build.sbt (hunk 2 above)
Comment (SuperIzya) [reply build.sbt:67]: :+1:

File: scalastyle-config.xml (hunk 8 above)
Comment (volod) [reply scalastyle-config.xml:106]: I'm not sure that numbers in methods names is a good approach. May be numbers can be allowed only for traits for special cases like function42. Suggestions

File: cats/src/scala/land/vola/cats/SparkTasks.scala (hunk 3 above)
Comment (SuperIzya) [reply cats/src/scala/land/vola/cats/SparkTasks.scala:10]: Nvm. My mistake

File: cats/src/scala/land/vola/cats/SparkTasks.scala (hunk 4 above)
Comment (SuperIzya) [reply cats/src/scala/land/vola/cats/SparkTasks.scala:13]: And here...

File: scalastyle-config.xml (hunk 5 above)
Comment (SuperIzya) [reply scalastyle-config.xml:39]: What about arity? Classes like `Function1`?

File: scalastyle-config.xml (hunk 9 above)
Comment (volod) [reply scalastyle-config.xml:117]: AtEof. Suggestion?

File: scalastyle-config.xml (hunk 8 above)
Comment (SuperIzya) [reply scalastyle-config.xml:106]: Ok. Let's leave it to traits for now. If and when we need it we will change the settings.

File: scalastyle-config.xml (hunk 9 above)
Comment (SuperIzya) [reply scalastyle-config.xml:117]: Should be

**Review by SuperIzya — APPROVED** [review]

**Pull request merged**